kani_metadata/src/lib.rs

```diff
@@ -1,17 +1,21 @@
 // Copyright Kani Contributors
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 
-mod harness;
-mod vtable;
+use serde::{Deserialize, Serialize};
 
+pub use artifact::ArtifactType;
 pub use harness::*;
 pub use vtable::*;
 
-use serde::{Deserialize, Serialize};
+pub mod artifact;
+mod harness;
+mod vtable;
 
 /// The structure of `.kani-metadata.json` files, which are emitted for each crate
-#[derive(Debug, Clone, Serialize, Deserialize)]
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
```

Comment:
I agree in principle but not in practice. I don't think there should be a default for these two structures because they should reflect existing data. That said, we do create mock versions of them and using Default just make the mocking code simpler.

I wonder if most of the mocking code will go away with the removal of `--dry-run`.